src/ray/core_worker/core_worker.cc

```diff
@@ -1347,6 +1351,32 @@ void CoreWorker::HandlePushTask(const rpc::PushTaskRequest &request,
     return;
   }
 
+  if (RayConfig::instance().gcs_service_enabled() &&
+      request.task_spec().type() == TaskType::ACTOR_CREATION_TASK) {
+    // When GCS server recovers from a failure, it will re-send the actor creation task to
+    // this core worker. In this case, we can just ignore this actor creation task.
+    // For more details please see the protocol of actor management based on gcs.
+    // https://docs.google.com/document/d/1EAWide-jy05akJp6OMtDn58XOK7bUyruWMia4E-fV28/edit?usp=sharing
+    ActorID current_actor_id;
+    WorkerID current_worker_id;
+    {
+      absl::MutexLock lock(&mutex_);
+      current_actor_id = actor_id_;
+      current_worker_id = worker_context_.GetWorkerID();
+    }
+    if (!current_actor_id.IsNil()) {
+      auto to_be_created_actor_id =
+          ActorID::FromBinary(request.task_spec().actor_creation_task_spec().actor_id());
+      RAY_CHECK(current_actor_id == to_be_created_actor_id);
+      RAY_LOG(WARNING)
+          << "Ignoring duplicated actor creation task for actor "
+          << to_be_created_actor_id << " in worker " << current_worker_id
+          << ". This is mostly likely because GCS was recovered from failure.";
+      send_reply_callback(Status::OK(), nullptr, nullptr);
```

Comment:
> I'm still confused about why this doesn't execute the actor creation task.

When both Raylet and Worker are working normally, if Actor1 has been created on the Worker1, when GCS Server is restarted, it will try to create Actor1 on the Worker1 again. At this time, the Worker1 should not create this Actor1 again. It just needs to tell GCS Server that the Actor1 has been created successfully.

Reply:
But this PR should not include any logic involving a persistent GCS server. So I would remove this for now from this PR, especially since it's not being covered by any e2e tests.

Reply:
> How does the GCS service know which worker the actor was started on (or it should start it on)? Is this persisted to the GCS? Looking in the design doc - is this in the second FlushActorRequest?

If the GCS Server restart, it will lease worker again from raylet which records the relationship between actor and leased worker, the raylet will reply the leased worker to the GCS Server, and then GCS Server will send the CreateActor request to the worker again.  

Anyway, I will remove this logic from this PR and add it back when persist storage is ready.